test/functional/eval/fnamemodify_spec.lua

```diff
@@ -42,14 +51,15 @@ describe('fnamemodify()', function()
     local filename = "src/version.c"
     local cwd = getcwd()
 
-    eq(cwd .. '/src/version.c', fnamemodify(filename, ':p'))
-    eq('src/version.c', fnamemodify(filename, ':p:.'))
-    eq(cwd .. '/src', fnamemodify(filename, ':p:h'))
-    eq(cwd .. '', fnamemodify(filename, ':p:h:h'))
-    eq('version.c', fnamemodify(filename, ':p:t'))
-    eq(cwd .. '/src/version', fnamemodify(filename, ':p:r'))
+    eq_slashconvert(cwd .. '/src/version.c', fnamemodify(filename, ':p'))
+
+    eq_slashconvert('src/version.c', fnamemodify(filename, ':p:.'))
+    eq_slashconvert(cwd .. '/src', fnamemodify(filename, ':p:h'))
+    eq_slashconvert(cwd .. '', fnamemodify(filename, ':p:h:h'))
+    eq_slashconvert('version.c', fnamemodify(filename, ':p:t'))
+    eq_slashconvert(cwd .. '/src/version', fnamemodify(filename, ':p:r'))
 
-    eq(cwd .. '/src/main.c', fnamemodify(filename, ':s?version?main?:p'))
+    eq_slashconvert(cwd .. '/src/main.c', fnamemodify(filename, ':s?version?main?:p'))
```

Comment:
`expected` also needs to be converted.

```
[  FAILED  ] test/functional\eval\fnamemodify_spec.lua @ 50: fnamemodify() handles examples from ":help filename-modifiers"
test/functional\eval\fnamemodify_spec.lua:16: Expected objects to be the same.
Passed in:
(string) 'C:\projects\neovim\src\version.c'
Expected:
(string) 'C:\projects\neovim/src/version.c'
stack traceback:
	test/functional\eval\fnamemodify_spec.lua:16: in function 'eq_slashconvert'
	test/functional\eval\fnamemodify_spec.lua:54: in function <test/functional\eval\fnamemodify_spec.lua:50>
```

Reply:
Ah of course. Just to check, would you be okay with/are you saying that we can ignore the type of slash for all of these tests? I guess we're primarily testing the path components we end up with and the types of slashes aren't important to us, and would only clutter the test case with `iswin()` checks.

I'll push up new tests with this change to kick off the test run - `alter_slashes` is exactly what I wanted, thanks!